src/cow-react/modules/swap/state/EthFlow/updaters/deadlineUpdater.tsx

```diff
@@ -0,0 +1,66 @@
+import { useEffect, useState } from 'react'
+import { useUserTransactionTTL } from 'state/user/hooks'
+import { useIsEthFlow } from '@cow/modules/swap/hooks/useIsEthFlow'
+
+const STORAGE_KEY = 'UserPreviousDeadline'
+// 10 minutes in SECONDS
+export const DEADLINE_LOWER_THRESHOLD_SECONDS = 600
+
+export function EthFlowDeadlineUpdater() {
+  const [mounted, setMounted] = useState(false)
+
+  // USER DEADLINE (IN SECONDS)
+  const [userDeadline, setUserDeadline] = useUserTransactionTTL()
+  const isEthFlow = useIsEthFlow()
+
+  // on updater mount
+  useEffect(() => {
+    const previousDeadline = _loadDeadline()
+
+    // only update if some storage amt exists
+    if (previousDeadline) {
+      setUserDeadline(previousDeadline)
+    }
+
+    setMounted(true)
+    // we only want this on mount
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [])
+
+  useEffect(() => {
+    if (isEthFlow) {
+      const deadlineLessThanThreshold = DEADLINE_LOWER_THRESHOLD_SECONDS > userDeadline
+      if (deadlineLessThanThreshold) {
+        console.log(
+          '[EthFlowDeadlineUpdater] - Setting user deadline to minimum threshold of 10 minutes',
```

Comment:
hardcoded threshold in message. The constant will probably change, but the message will not